File: U1/PROYECTO/01 SERVIDOR/Comercializadora_Restful_Java_GR01/src/main/java/ec/edu/monster/dto/CalculoFacturaDTO.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package ec.edu.monster.dto;

import java.io.Serializable;
import java.math.BigDecimal;
import java.util.ArrayList;
import java.util.List;

/**
 * DTO para respuesta de cálculo de factura (sin generarla)
 * Permite conocer el total antes de solicitar crédito
 */
public class CalculoFacturaDTO implements Serializable {

    private Boolean exitoso;
    private String mensaje;
    private BigDecimal total;
    private List<DetalleCalculoDTO> detalles;

    // Constructores
    public CalculoFacturaDTO() {
        this.detalles = new ArrayList<>();
    }

    public CalculoFacturaDTO(Boolean exitoso, String mensaje, BigDecimal total, List<DetalleCalculoDTO> detalles) {
        this.exitoso = exitoso;
        this.mensaje = mensaje;
        this.total = total;
        this.detalles = detalles != null ? detalles : new ArrayList<>();
    }

    // Getters y Setters
    public Boolean getExitoso() {
        return exitoso;
    }

    public void setExitoso(Boolean exitoso) {
        this.exitoso = exitoso;
    }

    public String getMensaje() {
        return mensaje;
    }

    public void setMensaje(String mensaje) {
        this.mensaje = mensaje;
    }

    public BigDecimal getTotal() {
        return total;
    }

    public void setTotal(BigDecimal total) {
        this.total = total;
    }

    public List<DetalleCalculoDTO> getDetalles() {
        return detalles;
    }

    public void setDetalles(List<DetalleCalculoDTO> detalles) {
        this.detalles = detalles;
    }
}

File: U1/EXAMEN_SOAP_DOTNET_GR01/02 CLICON/Comer_CliCon_SOAP_DotNet_GR01/Comer_CliCon_SOAP_DotNet_GR01/Models/CalculoFacturaDTO.cs
using System.Runtime.Serialization;

namespace Comer_CliCon_SOAP_DotNet_GR01.Models
{
    /// <summary>
    /// DTO para respuesta de cálculo de factura (sin generarla)
    /// Permite conocer el total antes de solicitar crédito
    /// </summary>
    [DataContract(Namespace = "http://schemas.datacontract.org/2004/07/Comercializadora_Soap_DotNet_GR01.DTOs")]
    public class CalculoFacturaDTO
    {
        [DataMember(Order = 1)]
        public bool Exitoso { get; set; }

        [DataMember(Order = 2)]
        public string Mensaje { get; set; } = string.Empty;

        [DataMember(Order = 3)]
        public decimal Total { get; set; }

        [DataMember(Order = 4)]
        public List<DetalleCalculoDTO> Detalles { get; set; } = new();
    }

    [DataContract(Namespace = "http://schemas.datacontract.org/2004/07/Comercializadora_Soap_DotNet_GR01.DTOs")]
    public class DetalleCalculoDTO
    {
        [DataMember]
        public int ProductoId { get; set; }

        [DataMember]
        public string NombreProducto { get; set; } = string.Empty;

        [DataMember]
        public int Cantidad { get; set; }

        [DataMember]
        public decimal PrecioUnitario { get; set; }

        [DataMember]
        public decimal Subtotal { get; set; }
    }
}
